src/main/java/domain/Movements.java

```diff
@@ -0,0 +1,54 @@
+package domain;
+
+import domain.piece.Piece;
+import java.util.List;
+
+public class Movements {
+    private final List<Movement> movements;
+
+    public Movements(List<Movement> movements) {
+        this.movements = movements;
+    }
+
+    public boolean canMoveFromTo(Position startPosition, Position endPosition) {
+        return movements.stream()
+                .anyMatch(movement -> movement.isValidMove(startPosition, endPosition));
+    }
+
+    public List<Position> findIntermediatePositions(Position startPosition, Position endPosition) {
+        return movements.stream()
+                .filter(movement -> movement.isValidMove(startPosition, endPosition))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("지정한 포지션으로 이동할 수 없습니다."))
+                .findIntermediatePositions(startPosition, endPosition);
+    }
+
+    public boolean canMovePieceToPosition(Piece target, Position expectedPosition, List<Piece> alivePieces) {
+        if (canNotMoveTargetTo(target, expectedPosition)) {
+            return false;
+        }
+        if (hasPieceAtIntermediatePositions(target, expectedPosition, alivePieces)) {
+            return false;
+        }
+        return true;
+    }
```

Comment:
아래와 같이 변경하면 더 간결하고 `canNotMoveTargetTo` 메소드도 필요 없어질 것 같습니다. 

```suggestion
 public boolean canMovePieceToPosition(Piece target, Position expectedPosition, List<Piece> alivePieces) {
    return this.canMoveFromTo(target.getPosition(), expectedPosition) &&
            !hasPieceAtIntermediatePositions(target, expectedPosition, alivePieces);
}
```

Reply:
부정문을 제거하기 위해 `canNotMoveTargetTo` 메소드를 따로 생성했는데, `canMoveFromTo`를 이용하여 작성한 코드가 더 읽기 편한 것 같네요 🥲 `if`분기를 제거하니, 더욱 간결해진 것 같아요. 

위 상황은 아니지만, 조건 분기가 더욱 많아진다면(3~4개의 조건 분기) `&&`를 이용하여 코드를 보는 입장에서 조건을 이해하는 게 어려울 것 같은데, 그런 경우는 따로 if문을 통해 작성하는 게 좋을까요? 물론 지금의 경우에는 피케이가 작성해주신 방법이 더욱 가독성이 좋은 것 같아요!

Reply:
조건이 길어져서 가독성이 떨어질 경우 해당 조건을 변수로 추출하는 것이 좋습니다.
변수 이름만 보고도 그 조건의 의미를 파악할 수 있게 지정하면 코드를 보는 입장에서도 조건을 더 잘 이해할 수 있으니까요.

Reply:
여러 조건들을 변수로 두어 **이름을 통해 어떤 의도를 가지고 있는 지** 나타내면 이후 내부 조건문을 읽을 때 쉽게 이해할 수 있을 것 같네요. 🤔 답변 감사해요 피케이 😃